app/Http/Controllers/Consumables/BulkConsumablesController.php

```diff
@@ -0,0 +1,195 @@
+                    $this->update_array['company_id'] =  $request->input('company_id');
+                    if ($request->input('company_id')=="clear") {
+                        $this->update_array['company_id'] = null;
+                    }
+                }
+                
+                if ($request->filled('qty')) {
+                    $this->update_array['qty'] = $request->input('qty');
+                }
+                
+                if ($request->filled('min_amt')) {
+                    $this->update_array['min_amt'] = $request->input('min_amt');
+                }
+                
+                if ($request->filled('category_id')) {
+                    $this->update_array['category_id'] = $request->input('category_id');
+                }
+
+                if ($request->filled('location_id')) {
+                    $this->update_array['location_id'] = $request->input('location_id');
+                }
+                
+                if ($request->filled('manufacturer_id')) {
+                    $this->update_array['manufacturer_id'] = $request->input('manufacturer_id');
+                }
+                
+                \Log::debug(implode(", ", $this->update_array));
+                \Log::debug(implode(", ", array_keys($this->update_array)));
+                DB::table('consumables')
+                    ->where('id', $consumableId)
+                    ->update($this->update_array);
+            } // endforeach
+            return redirect()->route("consumables.index")->with('success', trans('admin/consumables/message.update.success'));
+        // no values given, nothing to update
+        }
+        return redirect()->route("consumables.index")->with('warning', trans('admin/consumables/message.update.nothing_updated'));
+
+    }
+
+    /**
+     * Array to store update data per item
+     * @var Array
+     */
+    private $update_array;
+
+    /**
+     * Adds parameter to update array for an item if it exists in request
+     * @param  String $field field name
+     * @return BulkAssetsController Model for Chaining
+     */
+    protected function conditionallyAddItem($field)
+    {
+        if(request()->filled($field)) {
+            $this->update_array[$field] = request()->input($field);
+        }
+        return $this;
+    }
+
+    /**
+     * Save bulk deleted.
+     *
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * @param Request $request
+     * @return View
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     * @internal param array $assets
+     * @since [v2.0]
+     */
+    public function destroy(Request $request)
+    {
+        $this->authorize('delete', Consumable::class);
+
+        if ($request->filled('ids')) {
+            $consumables = Consumable::find($request->get('ids'));
+            foreach ($consumables as $consumable) {
+                $update_array['deleted_at'] = date('Y-m-d H:i:s');
+
+                DB::table('consumables')
```

Comment:
I think we can just use the `->delete()` method here?

Also how are we accounting for consumables that have been checked out already?